client/src/utils/ajax.ts

```diff
@@ -51,16 +55,93 @@ async function request<T>(
 /** GET **/
 
 interface SessionUser {
-  user: UserType;
+  user: { [username: string]: UserType };
   sessionMeta: { activeDonations: number };
-  result: string;
 }
-export function getSessionUser(): Promise<SessionUser> {
-  return get('/user/get-session-user');
+
+type challengeFilesForFiles = {
+  files: Array<Omit<ChallengeFile, 'fileKey'> & { key: string }>;
+} & Omit<CompletedChallenge, 'challengeFiles'>;
+
+type ApiSessionResponse = Omit<SessionUser, 'user'>;
+type ApiUser = {
+  user: {
+    [username: string]: ApiUserType;
+  };
+  result: string;
+};
+
+type ApiUserType = Omit<UserType, 'completedChallenges'> & {
+  completedChallenges: challengeFilesForFiles[];
+};
+
+type UserResponseType = {
+  user: { [username: string]: UserType };
+  result: string;
+};
+
+function parseApiResponseToClientUser<T>(
+  data: ApiUser | (T & ApiUser)
+): UserResponseType {
+  const userData = data.user[data.result];
+  let completedChallenges: CompletedChallenge[] = [];
+  if (userData) {
+    completedChallenges =
+      userData.completedChallenges.reduce(
```

Comment:
```suggestion
      userData.completedChallenges?.reduce(
```
Handling the missing challenges.